pod-resizer/nanny/k8s_client.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nanny
+
+import (
+	"fmt"
+
+	api "k8s.io/kubernetes/pkg/api"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/release_1_2"
+)
+
+type KubernetesClient interface {
+	CountNodes() (uint64, error)
+	ContainerResources() (*apiv1.ResourceRequirements, error)
+	UpdateDeployment(resources *apiv1.ResourceRequirements) error
+}
+
+type k8s_client_impl struct {
+	namespace, deployment, pod, container string
+	clientset                             *release_1_2.Clientset
+}
+
+func (k *k8s_client_impl) CountNodes() (uint64, error) {
+	opt := api.ListOptions{Watch: false}
+
+	nodes, err := k.clientset.CoreClient.Nodes().List(opt)
+	if err != nil {
+		return 0, err
+	}
+	return uint64(len(nodes.Items)), nil
+}
+
+func (k *k8s_client_impl) ContainerResources() (*apiv1.ResourceRequirements, error) {
+	pod, err := k.clientset.CoreClient.Pods(k.namespace).Get(k.pod)
+
+	if err != nil {
+		return nil, err
+	}
+	for _, container := range pod.Spec.Containers {
+		if container.Name != k.container {
```

Comment:
it's a bit strange that you use the opposite pattern here and in UpdateDeployment below. This would be simpler as

```
if container.Name == k.container {
  return &container.Resources, nil
}
```